sympy/combinatorics/fp_groups.py

```diff
@@ -277,6 +282,50 @@ def __str__(self):
     __repr__ = __str__
 
 
+class FpSubgroup(DefaultPrinting):
+    '''
+    The class implementing a subgroup of an FpGroup (only finite index subgroups are
+    supported at this point). This is to be used if one wishes to check if an element
+    of the original group belongs to the subgroup
+
+    '''
+    def __init__(self, G, gens):
+        super(FpSubgroup,self).__init__()
+        self.parent = G
+        self.generators = list(set([g for g in gens if g != G.identity]))
+        self.C = None
+
+    def __contains__(self, g):
+        if not g in self.parent:
+            raise ValueError
+        gens = g.contains_generators()
+        if any([g not in self.generators for g in gens]):
```

Comment:
> So there need not be any generators of the parent group.

True, I thought this when I was writing it and decided to come back later to change this to a set of parent generators involved in the words of `self.generators`, but I never did come back. Looking at it now, I'm not sure this check would do much for performance anyway so, maybe, it's not necessary at all.